pkg/kv/kvserver/batcheval/split_trigger_datadriven.go

```diff
@@ -0,0 +1,461 @@
+						Term:          10,
+						MinExpiration: hlc.Timestamp{WallTime: 100},
+					}
+				case "epoch":
+					lease = roachpb.Lease{
+						Replica: *targetReplica,
+						Epoch:   20,
+					}
+				case "expiration":
+					lease = roachpb.Lease{
+						Replica:    *targetReplica,
+						Expiration: &hlc.Timestamp{WallTime: 300},
+					}
+				default:
+					return fmt.Sprintf("unknown lease type: %s", leaseType)
+				}
+				rs.lease = lease
+				return fmt.Sprintf("set lease for range %d replica %d: %s", rangeID, replicaNodeID, leaseType)
+
+			case "print-lease":
+				var rangeID int
+				d.ScanArgs(t, "range-id", &rangeID)
+				rs := tc.mustGetRangeState(t, rangeID)
+
+				lease := rs.lease
+				var leaseType string
+				if lease.Epoch != 0 {
+					leaseType = "epoch"
+				} else if lease.Expiration != nil {
+					leaseType = "expiration"
+				} else {
+					leaseType = "leader-lease"
+				}
+				return fmt.Sprintf("range %d: leaseholder NodeID=%d, type=%s",
+					rangeID, lease.Replica.NodeID, leaseType)
+
+			case "print-range-state":
+				var sb strings.Builder
+				if len(tc.ranges) == 0 {
+					return "no ranges in test context"
+				}
+				// Sort range IDs for consistent output.
+				var rangeIDs []int
+				for rangeID := range tc.ranges {
+					rangeIDs = append(rangeIDs, rangeID)
+				}
+				sort.Ints(rangeIDs)
+
+				for _, rangeID := range rangeIDs {
+					rs := tc.ranges[rangeID]
+					sb.WriteString(fmt.Sprintf("range %d: [%s,%s) replicas=",
+						rangeID, rs.desc.StartKey, rs.desc.EndKey))
+
+					for i, replica := range rs.desc.InternalReplicas {
+						if i > 0 {
+							sb.WriteString(",")
+						}
+						sb.WriteString(fmt.Sprintf("n%d(r%d)", replica.NodeID, replica.ReplicaID))
+					}
+					sb.WriteString("\n")
+				}
+				return sb.String()
+
+			case "run-split-trigger":
+				var rangeID int
+				d.ScanArgs(t, "range-id", &rangeID)
+				split, ok := tc.splits[rangeID]
+				if !ok {
+					return fmt.Sprintf("no split trigger for range-id %d", rangeID)
+				}
+				rs := tc.mustGetRangeState(t, rangeID)
+				desc := rs.desc
+				db := storage.NewDefaultInMemForTesting()
+				defer db.Close()
+				batch := db.NewBatch()
+				defer batch.Close()
+
+				rec := (&MockEvalCtx{
+					ClusterSettings:        tc.st,
+					Desc:                   &desc,
+					Clock:                  tc.clock,
+					AbortSpan:              rs.abortspan,
+					LastReplicaGCTimestamp: rs.lastGCTimestamp,
+					RangeLeaseDuration:     tc.rangeLeaseDuration,
+				}).EvalContext()
+
+				// Write the range state that will be consulted and copied
+				// during the split.
+				sl := stateloader.Make(desc.RangeID)
+				err := sl.SetLease(ctx, batch, nil, rs.lease)
+				require.NoError(t, err)
+				err = sl.SetGCThreshold(ctx, batch, nil, &rs.gcThreshold)
+				require.NoError(t, err)
+				err = sl.SetGCHint(ctx, batch, nil, &rs.gcHint)
+				require.NoError(t, err)
+				err = sl.SetVersion(ctx, batch, nil, &rs.version)
+				require.NoError(t, err)
```

Comment:
I think we can refactor `splitTriggerHelper` in such a way that the input is pre-loaded and passed in as a struct. Then this piece of code (as well as a [similar](https://github.com/cockroachdb/cockroach/blob/e8afcff2c4934379c54ae401b61f3271a66c48cb/pkg/kv/kvserver/batcheval/cmd_end_transaction_test.go#L2194-L2209) one in another test) can be reused or not needed. It can also be shared with the non-test code.